src/canillita.erl

```diff
@@ -13,7 +13,9 @@ start() -> {ok, _} = application:ensure_all_started(canillita).
 stop() -> ok = application:stop(canillita).
 
 -spec start(Type::application:start_type(), Args::any()) -> {ok, pid()}.
-start(_Type, _Args) -> {ok, self()}.
+start(_Type, _Args) ->
+  {ok, _Pid} = canillita_sup:start_link(),
+  {ok, self()}.
```

Comment:
You should return the pid from `canillita_sup`. Returning `self()` was a _trick_ to comply with the OTP behaviour while doing nothing. What's expected for OTP applications is to return the pid of their main supervisor.
The expected code looks like…
```erlang
start(_Type, _Args) -> canillita_sup:start_link().
```